src/frontend/src/flows/verifiableCredentials/index.ts

```diff
@@ -85,9 +86,8 @@ const verifyCredentials = async ({
   const issuerCanisterId = lookedUp.ok;
 
   // If the RP provided a canister ID, check that it matches what we got
-  if (nonNullish(expectedIssuerCanisterId_)) {
-    const expectedCanisterId = expectedIssuerCanisterId_?.toText();
-    if (expectedCanisterId !== issuerCanisterId) {
+  if (nonNullish(expectedIssuerCanisterId)) {
+    if (expectedIssuerCanisterId !== issuerCanisterId) {
```

Comment:
It used to be `string` (hence the `expectedIssuerCanisterId_?.toText()` call). This PR changes it to be properly typed as `Principal` so now we can compare directly instead of transforming `expectedIssuerCanisterId` to text first.

Reply:
Are you aware that it's risky to compare objects with `!==` or `===`.

It checks whether they reference the same object, not whether the object contain the same.

For example, `({}) === ({})` is `false`.

Reply:
That makes sense, thanks!